walnut/validate/validate.go

```diff
@@ -0,0 +1,100 @@
+/*
+ * === This file is part of ALICE O² ===
+ *
+ * Copyright 2020 CERN and copyright holders of ALICE O².
+ * Author: Ayaan Zaidi <azaidi@cern.ch>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * In applying this license CERN does not waive the privileges and
+ * immunities granted to it by virtue of its status as an
+ * Intergovernmental Organization or submit itself to any jurisdiction.
+ */
+
+package validate
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	"github.com/AliceO2Group/Control/walnut/app"
+
+	"github.com/xeipuuv/gojsonschema"
+	"gopkg.in/yaml.v2"
+)
+
+// Template accepts a filename and format then validate against the schema specified (either workflow or task)
+func Template(filename string, format string) {
```

Comment:
I suggest we call it `CheckSchema` or `Check` or `SyntaxCheck` or even `Validate` (your pick).
`Template` has a specific meaning, and there is no template processing (i.e. resolution of `{{  }}` blocks) going on here, just pure YAML validation.

Reply:
I had avoided using `Validate`,  as calling this function would result in `validate.Validate()`. I think `CheckSchema` works well with `validate.CheckSchema()`. Although I would've preferred `validate.Schema()` but that's a bit repetitive.

Reply:
`validate.CheckSchema()` is ok for me.

If you prefer something shorter, you could also consider renaming the `validate` package to `schema`, which gets you `schema.Validate()`.

Reply:
How about merging the two packages `schemata` and `validate` into just `schemata`? I don't think we need the `validate` function to be separate from the package that generates the schema.

Reply:
Sure, fine with me.